src/solvers/dg_p4est/containers_2d.jl

```diff
@@ -145,39 +198,27 @@ function init_interfaces_iter_face_inner(info, sides, interfaces, interface_id,
   return nothing
 end
 
-function init_interfaces!(interfaces, mesh::P4estMesh{2})
-  # Let p4est iterate over all interfaces and call init_interfaces_iter_face
-  iter_face_c = @cfunction(init_interfaces_iter_face, Cvoid, (Ptr{p4est_iter_face_info_t}, Ptr{Cvoid}))
-  user_data = [interfaces, 1, mesh]
+function init_interfaces!(interfaces, mesh::P4estMesh)
+  # Let p4est iterate over all interfaces and call init_surfaces_iter_face
+  iter_face_c = @cfunction(init_surfaces_iter_face,
+                           Cvoid, (Ptr{p4est_iter_face_info_t}, Ptr{Cvoid}))
+  user_data = InitSurfacesIterFaceUserData(
+    interfaces,
+    nothing, # mortars
+    nothing, # boundaries
+    mesh)
 
   iterate_faces(mesh, iter_face_c, user_data)
```

Comment:
Would it make sense to initialize all containers at once here? Are interfaces, boundaries, or mortars ever initialized alone? Would there be a significant performance benefit by only iterating once?

Reply:
This `init_interfaces!` is now only called when constructing the `InterfaceContainerP4est`, which is not really performance-critical since it's only called once before starting a simulation. When doing AMR, all surfaces are re-initialized together by calling `init_surfaces!`.

Reply:
Ah, okay. That's cool.